DevEdu.API/DevEdu.DAL/Repositories/UserRepository.cs

```diff
@@ -45,19 +44,59 @@ public int AddUser(UserDto user)
 
         public UserDto SelectUserById(int id)
         {
-            return _connection.QuerySingleOrDefault<UserDto>(
+            UserDto result = default;
+            return _connection
+                .Query<UserDto, City, Role, UserDto>(
                 _userSelectByIdProcedure,
+                (user, city, role) =>
+                {
+                    if (result == null)
+                    {
+                        result = user;
+                        result.CityId = city;
+                        result.Roles = new List<Role> { role };
+                    }
+                    else
+                    {
+                        result.Roles.Add(role);
+                    }
+                    return result;
+                },
                 new { id },
-            commandType: CommandType.StoredProcedure);
+                splitOn: "id",
+            commandType: CommandType.StoredProcedure)
+                .FirstOrDefault();
         }
 
-        public List<UserDto> SelectUsers()
+        public Dictionary<int, UserDto> SelectUsers()
         {
-            return _connection
-                .Query<UserDto>(
-                _userSelectAllProcedure,
-            commandType: CommandType.StoredProcedure)
-                .ToList();
+            var UserDictionary = new Dictionary<int, UserDto>();
+
+            _connection
+               .Query<UserDto, City, Role, UserDto>(
+               _userSelectAllProcedure,
+               (user, city, role) =>
+               {
+                   UserDto result;
+
+                   if (!UserDictionary.TryGetValue(user.Id, out result))
+                   {
+                       result = user;
+                       result.CityId = city;
+                       result.Roles = new List<Role>();
+                       UserDictionary.Add(user.Id, result);
+                   }
+                   else
+                   {
+                       result.Roles.Add(role);
+                   }
+                   return result;
+               },
+               splitOn: "Id",
+           commandType: CommandType.StoredProcedure)
+               .Distinct()
+               .ToList();
+            return UserDictionary;
```

Comment:
this dictionary is a helper
you shouldn't return it from the method
please use debug and look why do we have .Distinct() here